packages/rum-core/src/common/apm-server.js

```diff
@@ -81,11 +81,14 @@ class ApmServer {
       headers: {
         'Content-Type': 'application/x-ndjson'
       }
-    })
+    }).then(({ responseText }) => responseText)
   }
 
   _constructError(reason) {
     const { url, status, responseText } = reason
+    if (typeof status == 'undefined') {
```

Comment:
can you explain a bit on why we need this special case?

Reply:
This is when the` reason` is any thing but a XHR object, for example the `reason` could be an `Error` if there's a bug in our code that throws an exception. without this that exception is lost which makes it harder for us to debug.